sources/scraping/helpers.py

```diff
@@ -0,0 +1,159 @@
+        self.spider = spider
+        self.settings = settings
+        self.start_urls = start_urls
+        self.include_headers = include_headers
+
+    def run(self, *args: Any, **kwargs: AnyDict) -> None:
+        runner = CrawlerRunner(settings=self.settings)
+        runner.crawl(
+            self.spider,
+            queue=self.queue,
+            include_headers=self.include_headers,
+            name="scraping_spider",
+            start_urls=self.start_urls,
+            **kwargs,
+        )
+
+        d = runner.join()
+        d.addBoth(lambda _: reactor.stop())  # type: ignore[attr-defined]
+        reactor.run()  # type: ignore[attr-defined]
+
+
+class PipelineRunner(Runnable):
+    def __init__(self, pipeline: dlt.Pipeline) -> None:
+        self.pipeline = pipeline
+
+    def run(  # type: ignore[override]
+        self,
+        data: Any,
+        *args: Any,
+        write_disposition: t.Optional[TWriteDisposition] = "append",
+        **kwargs: Any,
+    ) -> None:
+        """You can use all regular dlt.pipeline.run() arguments
+
+        ```
+        destination: TDestinationReferenceArg = None,
+        staging: TDestinationReferenceArg = None,
+        dataset_name: str = None,
+        credentials: Any = None,
+        table_name: str = None,
+        write_disposition: TWriteDisposition = None,
+        columns: TAnySchemaColumns = None,
+        primary_key: TColumnNames = None,
+        schema: Schema = None,
+        loader_file_format: TLoaderFileFormat = None
+        ```
+        """
+
+        def run() -> None:
+            self.pipeline.run(data, write_disposition=write_disposition, **kwargs)
+
+        self.thread_runner = threading.Thread(target=run)
+        self.thread_runner.start()
+
+    def join(self) -> None:
+        self.thread_runner.join()
+
+
+# We need to do this because scrapy requires passing
+# module path instead of class type which it later
+# resolves internally. Why this is done this way
+# is because using inspect to get modulename etc.
+# will even more complicate things and ScrapingPipelineItem
+# at the moment is defined under `.scrapy.pipepline_item` is
+# enough for use just to use `__package__` shortcut and
+# manually construct module path. So if you move it somehwere else
+# please provide it to runner.
+ScrapyPipelineItemModule = f"{__package__}.scrapy.pipeline_item.ScrapingPipelineItem"
+
+
+@with_config(sections=("sources", "scraping"), spec=ScrapingConfig)
+def create_pipeline_runner(
+    pipeline: dlt.Pipeline,
+    queue: t.Optional[BaseQueue[T]] = None,
+    queue_size: int = dlt.config.value,
+    spider: t.Type[DltSpider] = None,
+    start_urls: t.List[str] = dlt.config.value,
+    include_headers: t.Optional[bool] = dlt.config.value,
+) -> t.Tuple[PipelineRunner, ScrapyRunner, t.Callable[[], None]]:
+    if queue is None:
+        queue = BaseQueue(maxsize=queue_size)
+
+    settings = {
+        "ITEM_PIPELINES": {ScrapyPipelineItemModule: 100},
```

Comment:
what this `100` here means?

Reply:
It defines the order of execution (calls) to pipeline items
```
The integer values you assign to classes in this setting determine the order in which they run: items go through from lower valued to higher valued classes. It’s customary to define these numbers in the 0-1000 range.
```